src/IBM.Cloud.SDK.Core/Util/TokenManager.cs

```diff
@@ -99,9 +99,9 @@ public void SetAccessToken(string iamAccessToken)
         /// Request an IAM token using an API key.
         /// </summary>
         /// <returns>An IamTokenData object containing the IAM token.</returns>
-        private IamTokenData RequestToken()
+        private DetailedResponse<IamTokenData> RequestToken()
```

Comment:
I think I'm having a little trouble understanding the responses completely but just want to verify. Are the return types consistent here? This method is returning a `DetailedResponse<IamTokenData>`, but line 127 seems to pull out and return the `Result`, which I'd expect to be an instance of `IamTokenData`. Then, up on line 73, we get the `.Result` again. However, in case of null in `RequestToken`, we return a full `DetailedResponse<IamTokenData>`.

Does that work out okay and I'm just misunderstanding it? Sorry if that was confusing to follow.

Reply:
Yeah the network calls always return `DetailedResponse<T>` where `T` is whatever the return type is. In 127 we are calling the `As` method on the request which sticks whatever return type you have into a `DetailedResponse` https://github.com/IBM/dotnet-sdk-core/blob/feature-ibm-response/src/IBM.Cloud.SDK.Core/Http/Request.cs#L121

The `Result` in line 73 is the `Result` property of `DetailedResponse` which in this case is `IamTokenData`. 

Everything works fine now. A bunch of this might change this week because I'm trying to add async support for the major release.

Reply:
Ah okay sounds good 👍  And if everything still works we should be good!